src/hotspot/share/classfile/dictionary.hpp

```diff
@@ -89,51 +88,34 @@ class Dictionary : public CHeapObj<mtClass> {
   void verify();
 
  private:
-  bool is_valid_protection_domain(JavaThread* current, Symbol* name,
+  bool is_in_package_access_cache(JavaThread* current, Symbol* name,
                                   Handle protection_domain);
-  void add_protection_domain(JavaThread* current, InstanceKlass* klass,
-                             Handle protection_domain);
+  void add_to_package_access_cache(JavaThread* current, InstanceKlass* klass,
+                                   Handle protection_domain);
 };
 
-// An entry in the class loader data dictionaries, this describes a class as
-// { InstanceKlass*, protection_domain_set }.
-
 class DictionaryEntry : public CHeapObj<mtClass> {
  private:
-  // Contains the set of approved protection domains that can access
-  // this dictionary entry.
-  //
-  // [Note that C.protection_domain(), which is stored in the java.lang.Class
-  // mirror of C, is NOT the same as PD]
-  //
-  // If an entry for PD exists in the list, it means that
-  // it is okay for a caller class to reference the class in this dictionary entry.
-  //
-  // The usage of the PD set can be seen in SystemDictionary::validate_protection_domain()
-  // It is essentially a cache to avoid repeated Java up-calls to
-  // ClassLoader.checkPackageAccess().
```

Comment:
This was a pretty helpful comment, especially that we're trying to avoid repeated Java upcalls.  Why not just rename validate_protection_domain here too.

Reply:
I updated the comment to talk about avoiding the Java upcalls and to reflect the new names.